tests/common/snappi_tests/traffic_generation.py

```diff
@@ -769,3 +779,737 @@ def verify_egress_queue_frame_count(duthost,
+        pcap_bytes = api.get_capture(request)
+        with open(snappi_extra_params.packet_capture_file + ".pcapng", 'wb') as fid:
+            fid.write(pcap_bytes.getvalue())
+
+    # Dump per-flow statistics for final rows
+    logger.info("Dumping per-flow statistics for final row")
+    m = iter_count
+    f_stats[m] = {'Date': datetime.now().strftime('%Y-%m-%d-%H-%M-%S')}
+    flow_metrics = fetch_snappi_flow_metrics(api, data_flow_names)
+    traf_metrics = StatViewAssistant(ixnet_rest_api, 'Traffic Item Statistics').Rows
+    tx_frame = sum([metric.frames_tx for metric in flow_metrics if metric.name in data_flow_names])
+    rx_frame = sum([metric.frames_rx for metric in flow_metrics if metric.name in data_flow_names])
+    f_stats[m]['tgen_tx_frames'] = tx_frame
+    f_stats[m]['tgen_rx_frames'] = rx_frame
+    f_stats = update_dict(m, f_stats, tgen_curr_stats(traf_metrics, flow_metrics, data_flow_names))
+    for dut, port in dutport_list:
+        f_stats = update_dict(m, f_stats, flatten_dict(get_interface_stats(dut, port)))
+        f_stats = update_dict(m, f_stats, flatten_dict(get_pfc_count(dut, port)))
+        f_stats = update_dict(m, f_stats, flatten_dict(get_queue_count_all_prio(dut, port)))
+
+    for lossless_prio in switch_tx_lossless_prios:
+        for dut, port in dutport_list:
+            logger.info('Queue count for DUT: {}, Port:{} is {}'.
+                        format(dut.hostname, port, (get_egress_queue_count(dut, port, lossless_prio)[0])))
+    flow_metrics = fetch_snappi_flow_metrics(api, all_flow_names)
+    time.sleep(10)
+
+    df = pd.DataFrame(f_stats)
+    df_t = df.T
+    df_t = df_t.reindex(sorted(df_t.columns), axis=1)
+    flow_list = []
+    all_flow_names = [flow.name for flow in config.flows]
+    for item in all_flow_names:
+        flow_list.append(item.replace(' ', '_').lower())
+    results = list(df_t.columns)
+    fname = fname + '-' + datetime.now().strftime('%Y-%m-%d-%H-%M')
+    with open(fname+'.txt', 'w') as f:
+        f.write('Captured data for {} iterations at {} seconds interval \n'.format(m, stats_interval))
+        test_stats = {}
+        test_stats['tgen_loss_pkts'] = 0
+        test_stats['tgen_lossy_rx_pkts'] = 0
+        test_stats['tgen_lossy_tx_pkts'] = 0
+        test_stats['tgen_lossless_rx_pkts'] = 0
+        test_stats['tgen_lossless_tx_pkts'] = 0
+        test_stats['tgen_rx_rate'] = 0
+        test_stats['tgen_tx_rate'] = 0
+        for flow in flow_list:
+            rx_rate = 0
+            tx_rate = 0
+            for item in results:
+                if (flow in item and item.split(flow)[1] == '_avg_latency_ns'):
+                    avg_latency = round(df_t[item].mean(), 2)
+                if (flow in item and item.split(flow)[1] == '_rx_pkts'):
+                    rx_pkts = df_t[item].max()
+                if (flow in item and item.split(flow)[1] == '_rx_pkts' and ('prio_3' in flow or 'prio_4' in flow)):
```

Comment:
we can use the lossless priority list and lossy priority list instead of hard code prio_3/prio_4.